packages/itwin/grouping-mapping-widget/src/widget/components/hooks/useMappingsOperations.ts

```diff
@@ -30,22 +31,82 @@ const fetchMappings = async (
   }
 };
 
+const fetchExtractionStatus = async (
+  iModelId: string,
+  getAccessToken: GetAccessTokenFn,
+  extractionClient: IExtractionClient,
+  setIconStatus: React.Dispatch<React.SetStateAction<"negative" | "positive" | "warning">>,
+  setIconMessage: React.Dispatch<React.SetStateAction<string>>,
+  setExtractionMessageData: React.Dispatch<React.SetStateAction<ExtractionMessageData[]>>
+) => {
+  setIconStatus("warning");
+  setIconMessage("Extraction status pending.");
+  try {
+    const accessToken = await getAccessToken();
+    const extractions = await extractionClient.getExtractionHistory(accessToken, iModelId);
```

Comment:
Since you are only interested in the latest extraction, you can pass the `top` value of `1` to `getExtractionHistory`. This will request the API only for the latest extraction history entry.

Reply:
I have updated this part.